components/support/ktx/src/main/java/mozilla/components/support/ktx/android/net/Uri.kt

```diff
@@ -53,3 +57,22 @@ fun Uri.sameSchemeAndHostAs(other: Uri) = scheme == other.scheme && host == othe
  * https://html.spec.whatwg.org/multipage/origin.html#same-origin
  */
 fun Uri.sameOriginAs(other: Uri) = sameSchemeAndHostAs(other) && port == other.port
+
+/**
+ * Indicate if the [this] uri is under the application private directory.
+ */
+fun Uri.isUnderPrivateAppDirectory(context: Context): Boolean {
+    return when (this.scheme) {
+        ContentResolver.SCHEME_FILE -> {
+            try {
+                val uriPath = path ?: return true
+                val uriCanonicalPath = File(uriPath).canonicalPath
```

Comment:
Not sure why..
but on my Android 11 emulator canonicalPath would not resolve the `data/user/0` symlink to `data/data`.
Don't have a physical device with A11 to test..